### ISSUE TITLE
Add test coverage for openlibrary/plugins/openlibrary/js/subjects.js

### ISSUE THREAD

**Issue by jdlrobson** [internetarchive/openlibrary#2043]
**Description**
The openlibrary/plugins/openlibrary/js/subjects.js file contains a lot of code that can likely be refactored - for instance slice and rendertag - however without tests this comes with risk.

Write tests for the functions in this page with an aim for at least 70% coverage to describe how this file behaves.

**Comment by jdlrobson** [comment]
Cc @koderjoker

**Comment by jdlrobson** [comment]
While I'm traveling (until 23rd), here's a few pointers that might cause problems and may help you!

1)  jQuery is not available in the test framework right now. We use $.log, $.getJSON and $.extend in the file so make sure you define a mock jquery with these methods within the test before running. The easiest way to do this would be inside the [setup/teardown functions](https://jestjs.io/docs/en/setup-teardown.html).
```
var $ = require( '../mockjQuery' );
beforeEach(() => {
  window.$ = $;
});

afterEach(() => {
  window.$ = undefined.
});
```
where mockjQuery looks like this:
```
module.exports = {
  log: ()=>{},
  getJSON: ()=>{},
 extend: () => Object.assign(arguments );
};
```

2) Stubs and mocks of data will be useful. I recommend we use `[sinon](https://sinonjs.org/)` library to do this. Add it to package.json and use require directly in any test where that might be useful. Wikipedia's mobile site uses sinon so the code examples [in their repo](https://github.com/wikimedia/mediawiki-extensions-MobileFrontend/blob/80b5740f6a826b40fb65ddd8a0b8fa037e6819b7/tests/node-qunit/mobile.startup/watchstar/Watchstar.test.js) will likely be helpful.

Hopefully that's enough to get you busy. I'm on UTC+8 at the moment so hopefully we can sync week of 23-26th.

**Comment by guyjeangilles** [comment]
@koderjoker is there anything blocking progress on this issue?

**Comment by koderjoker** [comment]
@guyjeangilles the following issue has been taken on by @jdlrobson for the time being.

**Comment by jdlrobson** [comment]
this was partially fixed in #2225. Our coverage is currently 40% of lines. Tests for lines 47,57,89-216  are still needed. 

Does anybody want to work on this?

**Comment by anishlukk123** [comment]
@jdlrobson  can  this be assigned to me

**Comment by jdlrobson** [comment]
@anishlukk123 a patch here would be appreciated assuming @mekarpeles @RayBB is available to review!

**Comment by RayBB** [comment]
Closed via #8783
